### ISSUE TITLE
Error: Failed to execute 'add' on 'DOMTokenList': The token provided ('button-md-twitter') contains HTML space characters, which are not valid in tokens

### ISSUE THREAD

**Issue by amreladawy** [ionic-team/ionic#12600]
**Ionic version:**  (check one with "x")
[ ] **1.x** (For Ionic 1.x issues, please use https://github.com/ionic-team/ionic-v1)
[ ] **2.x**
[x] **3.x**

**I'm submitting a ...**  (check one with "x")
[ ] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/
[x] Improvement

**Current behavior:**
Setting a color attribute to a value with extra space(s) causes the following error.

```
InvalidCharacterError: Failed to execute 'add' on 'DOMTokenList': 
The token provided ('button-md-twitter ') contains HTML space characters, which are not 
valid in tokens.
```

**Expected behavior:**
As an improvement, the white spaces should be trim from the attribute values.

**Steps to reproduce:**

http://plnkr.co/edit/EXwheDpNmvzcmAyNkN6r?p=preview

**Related code:**

```
<button ion-button  color="twitter ">
    <ion-icon name="logo-twitter"></ion-icon>
</button>
```

**Other information:**

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

```
    @ionic/cli-utils  : 1.7.0
    ionic (Ionic CLI) : 3.7.0

System:

    Android SDK Tools : 26.0.2
    Node              : v6.10.1
    OS                : Windows 10
    npm               : 5.3.0
```

**Comment by jgw96** [comment]
Hello, thanks for using Ionic. We will look into this.

**Comment by hamzahamidi** [comment]
I have the same issue here with this code           
 ```   <p ion-text color=" item.state ? 'danger' : 'secondary'" >{{item.state}}</p>```

**Comment by DavidTheProgrammer** [comment]
For anybody who finds this in future, I had the exact same problem. Turns out it's literally the extra whitespace in the string that is assigning the color. 

`color='twitter'` instead of `color='twitter '`

Note the extra white-space at the end. This drove me insane for a good half hour, I don't know where the white-space was coming from on my end, maybe _Prettier_ auto formats but yeah that's the problem. I assume it's the same for @hamzahamidi the white-space between the `=` and `item.state` in 

`color=" item.state...` might be causing the error.

And it's not an Ionic thing, it's an Angular thing, had this error using Angular Material and came across this issue while looking for a fix.

**Comment by amreladawy** [comment]
@DavidTheProgrammer yes,  already mentioned the cause and the solution in the original pot.
